mozilla-services/consvc-shepherd
# [AE-636] Fix mobile tiles preview

## References

https://mozilla-hub.atlassian.net/browse/AE-636

## Problem Statement

In testing AMP callbacks, Glenda noticed that locally she couldn't get test mobile tiles from MARS but the preview page showed that working fine.

It turned out that there were a couple problems:
* preview wasn't updated with mobile placement names
* the UAPI requests are not specifying the user agent, so MARS was always determining them to be a desktop client

This made the tiles returned always come from the desktop AMP endpoint, and then they were rendered with the styles appropriate for mobile.

## Proposed Changes
This adds the mobile tile placement names and uses the FormFactors to drive that condition. It also adds the User-Agent header to UAPI requests.

## Verification Steps
Ran locally and verified that dev UAPI and prod UAPI both returned ads for desktop and mobile. This didn't really confirm the code was correct since the preview page was already showing ads for those; but it did show some small differences.

AMP returns 2 tiles for prod mobile, so even if we request 3 we only get 2 back. This difference is visible now.
AMP didn't have test tiles set up for mobile when I started this PR -- the thing that made us realize this gap. While it was a WIP there was some time where dev mobile tiles were an empty and the desktop dev tiles were not. Glenda has reached out to AMP to ask about this, and as of this morning it looks like there are some tiles in both now.
